pkg/controller/scyllanodeconfig/status.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2021 ScyllaDB
+
+package scyllanodeconfig
+
+import (
+	"context"
+
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+func (sncc *Controller) calculateStatus(snt *scyllav1alpha1.ScyllaNodeConfig, daemonSets map[string]*appsv1.DaemonSet) *scyllav1alpha1.ScyllaNodeConfigStatus {
+	status := snt.Status.DeepCopy()
+	status.ObservedGeneration = snt.Generation
+	status.Stale = true
+
+	for dsName, ds := range daemonSets {
+		if snt.Name == dsName {
+			status.NumberDesired = ds.Status.DesiredNumberScheduled
+			status.NumberScheduled = ds.Status.CurrentNumberScheduled
+			status.NumberReady = ds.Status.NumberReady
+			status.Stale = ds.Status.ObservedGeneration < ds.Generation
```

Comment:
should this be more like OR the overwriting it?

Reply:
I don't get it, could you elaborate what you have in mind?